server/api.go

```diff
@@ -727,23 +727,31 @@ func (p *Plugin) createIssueComment(w http.ResponseWriter, r *http.Request, user
 		return
 	}
 
+	commentUsername := ""
 	commentUser, appErr := api.GetUser(post.UserId)
 	if appErr != nil {
 		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load post.UserID " + post.UserId + ": not found", StatusCode: http.StatusInternalServerError})
 		return
 	}
+	if info, err := p.getGitHubUserInfo(post.UserId); err != nil {
+		commentUsername = commentUser.Username
+	} else {
+		commentUsername = info.GitHubUsername
+	}
 
-	currentUser, appErr := api.GetUser(userID)
-	if appErr != nil {
-		writeAPIError(w, &APIErrorResponse{ID: "", Message: "failed to load current user", StatusCode: http.StatusInternalServerError})
+	currentUsername := ""
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
 		return
+	} else {
+		currentUsername = info.GitHubUsername
 	}
 
 	siteUrl := api.GetConfig().ServiceSettings.SiteURL
 
 	permalink := getPermaLink(*siteUrl, req.PostId, req.CurrentTeam)
 
-	permalinkMessage := fmt.Sprintf("*@%s attached a* [message](%s) *from @%s*\n", currentUser.Username, permalink, commentUser.Username)
+	permalinkMessage := fmt.Sprintf("*@%s attached a* [message](%s) *from @%s*\n", currentUsername, permalink, commentUsername)
```

Comment:
While you a touching this part of the code, I would like to think about a change unrelated to the original ticket: Does it make sense to reverse the order and post the original message first?

Before:
---
*@hanzei attached a* [message](0xc0002536a0/_redirect/pl/a3egsu1n9bd9jy98h9dhz3p4wo) *from @hanzei*
Some messsage

After:
--- 
Some message

*@hanzei attached a* [message](0xc0002536a0/_redirect/pl/a3egsu1n9bd9jy98h9dhz3p4wo) *from @hanzei*



Let me know what you think

Reply:
In my opinion, the message which'll be attached might be long, so the info (@abdulsmapara attached a [message](https://github.com/mattermost/mattermost-plugin-github/pull/256) from @hanzei) might go unseen at the end of the attached message; hence, it is good to keep it above the attached message.
Still, if the change needs to be done, Iet me know.

Reply:
I'm 0/5 on my suggestion. Let's leave it as it is for now.